exec/vector/src/main/java/org/apache/drill/exec/vector/complex/impl/RepeatedListReaderImpl.java

```diff
@@ -80,7 +79,10 @@ public void reset() {
 
   @Override
   public int size() {
-    return maxOffset - currentOffset;
+    if (currentOffset == NO_VALUES) {
+      return 0;
+    }
+    return maxOffset - currentOffset - 1;
```

Comment:
Consider:
`return currentOffset == NO_VALUES ? 0 : maxOffset - currentOffset - 1;`

Reply:
Introduce also `isNull()` method like in `RepeatedMapReaderImpl` class

Reply:
In my opinion usage ternary or full conditional construct doesn't impact code readability here. I'm confused by unclear naming of the fields and necessity to use 'magic number' along with the fields. @KazydubB  could you please rename or provide descriptive javadocs and constant for 1 to make offsets usage look clear ?

Reply:
@KazydubB It will be good to have it for consistency with `ReapetedMapReaderImpl`.
You can use parentheses. 
`return isNull() ? 0 : (maxOffset - currentOffset - 1);`
The final decision is yours

@ihuzenko Good note. I think it will be good to have a comment, why `-1` is need.

Reply:
I agree that this is not clear why the `-1` is added, so decided to decrement `maxOffset` value by `1` for the count to be an expected statement `maxOffset - currentOffset`.

Also did add a method to `RepeatedListReaderImpl` but named it `isEmpty()` (instead of `isNull()`) as it appears to fit better.